app/scripts/metamask-controller.js

```diff
@@ -1611,7 +1620,7 @@ export default class MetamaskController extends EventEmitter {
 
     function updatePublicConfigStore(memState) {
       const { chainId } = networkController.store.getState().provider;
-      if (memState.network !== 'loading') {
+      if (memState.networkStatus === NetworkStatus.Available) {
```

Comment:
There is a reference to `network` a few lines down from here, the second argument to `selectPublicState`, that should be updated to reference `networkId ?? 'loading'`